bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/CreateNewLogTest.java

```diff
@@ -93,16 +98,16 @@ public void testCreateNewLog() throws Exception {
         // Extracted from createNewLog()
         String logFileName = Long.toHexString(1) + ".log";
         File dir = ledgerDirsManager.pickRandomWritableDir();
-        LOG.info("Picked this directory: " + dir);
+        LOG.info("Picked this directory: {}", dir);
         File newLogFile = new File(dir, logFileName);
         newLogFile.createNewFile();
 
         EntryLogger el = new EntryLogger(conf, ledgerDirsManager);
         // Calls createNewLog, and with the number of directories we
         // are using, if it picks one at random it will fail.
-        el.createNewLog();
-        LOG.info("This is the current log id: " + el.getCurrentLogId());
-        assertTrue("Wrong log id", el.getCurrentLogId() > 1);
+        ((EntryLogManagerBase) el.entryLogManager).createNewLog(0L);
```

Comment:
- nit: provide a `getEntryLogManager` in EntryLogger. so we can mock it in future if needed.
- better to cast to `SingleEntryLogManager`. because you can expose `getCurrentLogId` in `SingleEntryLogManager`.

```
SingleEntryLogManager selm = (SingleEntryLogManager) (el.getEntryLogManager());
selm.createNewLog(0L);
assertTrue(selm.getCurrentLogId() > 1);
```

Reply:
1) I can add simple getter for EntryLogManager in EntryLogger, but wondering how would it help us in mocking if inside EntryLogger class direct reference of EntryLogger instance is used and using getter of EntryLogManager in all the places it is used in EntryLogger class is a stretch. 
2) if mocking is the requirement here then i can create initializeEntryLogManager method, which would initialize the EntryLogManager depending on the config, and for mocking purpose that method can be mocked to return what we want.
3) ok. in some places I wanted to test  getPreviousAllocatedEntryLogId, but it seems in this test it is not appropriate to change it to getPreviousAllocatedEntryLogId. will revert it back.